src/runtime/HalideRuntimeCuda.h

```diff
@@ -40,18 +40,17 @@ extern int halide_cuda_run(void *user_context,
  * must be NULL when this routine is called. This call can fail due to
  * being passed an invalid device pointer. The device and host dirty
  * bits are left unmodified. */
-extern int halide_cuda_wrap_device_ptr(void *user_context, struct halide_buffer_t *buf, uintptr_t device_ptr);
+extern int halide_cuda_wrap_device_ptr(void *user_context, struct halide_buffer_t *buf, uint64_t device_ptr);
 
 /** Disconnect this halide_buffer_t from the device pointer it was
```

Comment:
The device_interface isn't actually allocated, it just sets the device_interface pointer to the address of the const global device_interface struct for the API. The implementation just sets the device field and the device_interface field to the arguments, and calls use_module.

I'll update the comment.